core/src/main/java/org/apache/druid/timeline/partition/ShardSpec.java

```diff
@@ -123,6 +123,12 @@ default short getAtomicUpdateGroupSize()
   @JsonIgnore
   boolean possibleInDomain(Map<String, RangeSet<String>> domain);
 
+  /**
+   * Get the type name of this ShardSpec.
+   */
+  @JsonIgnore
+  String getType();
```

Comment:
This should have a default implementation that returns something like "unknown".  I don't know if anybody has implemented this interface in an extension, but it's technically possible and no reason to break their stuff if they did.

Reply:
Thanks for the review, @cheddar .

Yeah, I kind of struggled with this too. Did start with a default but then got rid of it.

Fixing it.

Reply:
Done.